### ISSUE TITLE
Point scheduler PDF link to the proposal site

### ISSUE THREAD

**Issue by simonfernandes** [UserOfficeProject/user-office-project-issue-tracker#854]
In the scheduler, where a user can view an experiment and follow a link to the proposal PDF, we should update that link from SharePoint to the new proposal site where applicable.

**Comment by georgeflecknell** [comment]
This issue needs to move to the Scheduler team. We are currently unable to generate direct links to PDFs.

The broken Scheduler link is this one:

![image](https://github.com/UserOfficeProject/user-office-project-issue-tracker/assets/75302934/400b41f5-7c1b-4cbd-81ae-2a0af5b470af)

This issue may be blocked by: [INSERT ISSUE NUMBER HERE ONCE RAISED - GENERATING DIRECT LINKS TO PDFs]

Emma Gozzard told us this is a priority though as it is annoying Instrument Scientists and regularly used by them in scheduling tasks. We might however have a workaround of using a search link such as: https://proposal.facilities.rl.ac.uk/Proposals?search=2310666 for newer proposal that appear in the proposal system.

**Comment by mutambaraf** [comment]
Scheduler should be able to download proposals with an API token this is the documentation https://user-office-project.atlassian.net/wiki/spaces/UOP/pages/88047617/Querying+proposal+data+through+the+GraphQL+API#Downloading-proposal-pdf-documents-using-API-token. @simonfernandes  should this not be closed from our side.

**Comment by simonfernandes** [comment]
The scheduler just displays a link that the user then accesses and downloads so I think we still need #855

**Comment by mutambaraf** [comment]
I think scheduler should do a request through the API. They are able to  this call they can do `  $.ajax({
            async: false,
            type: "POST",
            url: 'calendar.aspx/postponeExperimentPart',
            data: "{'experimentPartId':'" + id + "'}",
            contentType: "application/json; charset=utf-8",
            success: function (msg) {
                RebindAll();
            },
            error: function (a, b, c) {
                showFadingnotification('Unable to postpone the experiment part', 'alert-error');
            }
        });`

**Comment by simonfernandes** [comment]
I'm not quite sure what you mean still. Could you open an issue in https://github.com/isisbusapps/Schedule/ with more detail, then they can decide how they want to do this and what they need from us

**Comment by mutambaraf** [comment]
Scheduler can also point directly to a proposal using any of the following.
1. `/?token=<sessionid>&proposalid=<rbnumber>`
2. `/?proposalid=<rbnumber>`
3. `/?sessionId=<sessionid>&proposalid=<rbnumber>`

**Comment by simonfernandes** [comment]
@go994 Please could you provide the scheduler issue for this? Thanks

**Comment by go994** [comment]
> @go994 Please could you provide the scheduler issue for this? Thanks

https://github.com/isisbusapps/Schedule/issues/3618

**Comment by simonfernandes** [comment]
Closed via https://github.com/isisbusapps/Schedule/issues/3618
